plugins/device/transaction.go

```diff
@@ -51,3 +52,28 @@ func txnPrepareDevice(c transaction.TxnCtx) error {
 	}
 	return nil
 }
+
+func txnEditGroup(c transaction.TxnCtx) error {
+	var peerID string
+	var groupID string
+	if err := c.Get("peerid", peerID); err != nil {
+		c.Logger().WithError(err).WithField("PeerID", peerID).Error("Failed transaction, cannot find peer-id")
+		return err
+	}
+	if err := c.Get("groupid", groupID); err != nil {
+		c.Logger().WithError(err).WithField("GroupID", groupID).Error("Failed transaction, cannot find group-id")
+		return err
+	}
+	peerInfo, err := peer.GetPeer(peerID)
+	if err != nil {
+		c.Logger().WithError(err).WithField("peerid", peerID).Error("Peer ID not found in store")
+		return err
+	}
+	peerInfo.MetaData["_group"] = groupID
```

Comment:
This is duplicate work. See if you can re-use the edit metadata functionality implemented in the other PR.

Reply:
Edit Peer functionality can't be used. Because any group key starting with "_" is restricted with that API so that user configured Metadata will not override internal components Metadata.

Reply:
What do you think of this ? Restrict edit peer handler from editing metadata keys that begin with `_`. We can re-use the transaction step function.

Reply:
Makes sense.

Reply:
@prashanthpai I have made changes in the the peer api PR,once that is merged I will update this code. Please review https://github.com/gluster/glusterd2/pull/600